crates/driver/src/boundary/liquidity/zeroex.rs

```diff
@@ -1,11 +1,75 @@
 use {
-    crate::{infra, infra::Ethereum},
+    crate::{
+        domain::liquidity::{
+            self,
+            zeroex::{self, Order},
+        },
+        infra::{self, Ethereum},
+    },
+    anyhow::anyhow,
     ethrpc::current_block::CurrentBlockStream,
-    shared::{http_client::HttpClientFactory, zeroex_api::DefaultZeroExApi},
-    solver::{liquidity::zeroex::ZeroExLiquidity, liquidity_collector::LiquidityCollecting},
+    model::order::OrderKind,
+    shared::{
+        http_client::HttpClientFactory,
+        price_estimation::gas::GAS_PER_ZEROEX_ORDER,
+        zeroex_api::DefaultZeroExApi,
+    },
+    solver::{
+        liquidity::{zeroex::ZeroExLiquidity, LimitOrder},
+        liquidity_collector::LiquidityCollecting,
+    },
     std::sync::Arc,
 };
 
+pub fn to_domain(id: liquidity::Id, pool: LimitOrder) -> anyhow::Result<liquidity::Liquidity> {
+    let handler = pool
+        .settlement_handling
+        .as_any()
+        .downcast_ref::<solver::liquidity::zeroex::OrderSettlementHandler>()
+        .ok_or(anyhow!("not a zeroex::OrderSettlementHandler"))?
+        .clone();
```

Comment:
I tried to avoid that, but we need to provide the contract and order somehow. Currently, it's not possible to add zeroex-specific fields right to the [solver::LimitOrder](https://github.com/cowprotocol/services/blob/ebe471630c7df72a9c8ef3acbd7452098b989b1c/crates/solver/src/liquidity.rs#L177) since it's used with different settlement handlers. To address it, looks like the whole [solver::Liquidity](https://github.com/cowprotocol/services/blob/ebe471630c7df72a9c8ef3acbd7452098b989b1c/crates/solver/src/liquidity.rs#L44-L50) should be moved to the `driver` crate first.